src/bootlint.js

```diff
@@ -409,7 +409,9 @@ var LocationIndex = _location.LocationIndex;
         }
     });
     addLinter("E004", function lintNestedContainers($, reporter) {
-        var nestedContainers = $('.container, .container-fluid').children('.container, .container-fluid');
+        var nestedContainers = $('.container, .container-fluid').filter(function () {
+            return $(this).find('.navbar').length === 0;
```

Comment:
All this time I'd been fighting cheerio with a `.find(':not(.navbar)')` check. I'll make the change.